stackslib/src/net/api/tests/getclaritymarfvalue.rs

```diff
@@ -0,0 +1,203 @@
+        true,
+    );
+    requests.push(request);
+
+    // query non-existant var
+    let request = StacksHttpRequest::new_getclaritymarf(
+        addr.into(),
+        TrieHash::from_key(
+            "vm::ST2DS4MSWSGJ3W9FBC6BVT0Y92S345HY8N3T6AV7R.hello-world::1::does-not-exist",
+        ),
+        TipRequest::UseLatestAnchoredTip,
+        true,
+    );
+    requests.push(request);
+
+    // query non-existant contract
+    let request = StacksHttpRequest::new_getclaritymarf(
+        addr.into(),
+        TrieHash::from_key("vm::ST2DS4MSWSGJ3W9FBC6BVT0Y92S345HY8N3T6AV7R.does-not-exist::1::bar"),
+        TipRequest::UseLatestAnchoredTip,
+        true,
+    );
+    requests.push(request);
+
+    // query vm-account balance
+    let request = StacksHttpRequest::new_getclaritymarf(
+        addr.into(),
+        TrieHash::from_key("vm-account::ST2DS4MSWSGJ3W9FBC6BVT0Y92S345HY8N3T6AV7R::19"),
+        TipRequest::UseLatestAnchoredTip,
+        true,
+    );
+    requests.push(request);
+
+    let mut responses = test_rpc(function_name!(), requests);
+
+    // existing data
+    let response = responses.remove(0);
+    debug!(
+        "Response:\n{}\n",
+        std::str::from_utf8(&response.try_serialize().unwrap()).unwrap()
+    );
+
+    assert_eq!(
+        response.preamble().get_canonical_stacks_tip_height(),
+        Some(1)
+    );
+
+    let resp = response.decode_data_var_response().unwrap();
+    assert_eq!(resp.data, "0x0000000000000000000000000000000000");
+    assert!(resp.marf_proof.is_some());
+
+    // unconfirmed data
+    let response = responses.remove(0);
+    debug!(
+        "Response:\n{}\n",
+        std::str::from_utf8(&response.try_serialize().unwrap()).unwrap()
+    );
+
+    assert_eq!(
+        response.preamble().get_canonical_stacks_tip_height(),
+        Some(1)
+    );
+
+    let resp = response.decode_data_var_response().unwrap();
+    assert_eq!(resp.data, "0x0100000000000000000000000000000001");
+    assert!(resp.marf_proof.is_some());
+
+    // no such var
+    let response = responses.remove(0);
+    debug!(
+        "Response:\n{}\n",
+        std::str::from_utf8(&response.try_serialize().unwrap()).unwrap()
+    );
+
+    let (preamble, body) = response.destruct();
+    assert_eq!(preamble.status_code, 404);
+
+    // no such contract
+    let response = responses.remove(0);
+    debug!(
+        "Response:\n{}\n",
+        std::str::from_utf8(&response.try_serialize().unwrap()).unwrap()
+    );
+
+    let (preamble, body) = response.destruct();
+    assert_eq!(preamble.status_code, 404);
+
+    // vm-account blaance
+    let response = responses.remove(0);
+    debug!(
+        "Response:\n{}\n",
+        std::str::from_utf8(&response.try_serialize().unwrap()).unwrap()
+    );
+
+    let resp = response.decode_data_var_response().unwrap();
+    let balance = Value::try_deserialize_hex(&resp.data[2..], &TypeSignature::IntType, false);
+    assert_eq!(balance, Ok(Value::Int(256_000_000_000)));
```

Comment:
I'm not sure that 256M is the right expected value 🤔

Reply:
Everything looks good to me, but I'm also unclear how this balance reaches 256,000 STX.

Reply:
@jcnelson Does that make sense to you?